### PR TITLE
RavenDB-12133 Allow to add node to a cluster only if both nodes can e…

### PR DESCRIPTION
**Pull request by karmeli87**
…stablish connection to each other

Also the test connection button for the cluster will check both directions.

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs [hunk 1]
```diff
@@ -335,6 +336,16 @@ public async Task AddNode()
...
                 requestExecutor.DefaultTimeout = ServerStore.Engine.OperationTimeout;
 
+                // test connection to remote.
+                var result = await ServerStore.TestConnectionToRemote(nodeUrl, database: null);
+                if (result.Success)
+                {
+                    // test connection from remote to destination
+                    result = await ServerStore.TestConnectionFromRemote(requestExecutor, ctx, nodeUrl);
+                    if(result.Success == false)
+                        throw new InvalidOperationException(result.Error);
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs]: Here need to indicate that we were able to connect to the remote, and it wasn't able to connect to us

File: src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs [hunk 2]
```diff
@@ -335,6 +336,16 @@ public async Task AddNode()
             {
                 requestExecutor.DefaultTimeout = ServerStore.Engine.OperationTimeout;
 
+                // test connection to remote.
+                var result = await ServerStore.TestConnectionToRemote(nodeUrl, database: null);
+                if (result.Success)
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs]: where is the error handling on the `Success == false`?

File: src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs]: The `result.Error` already contains this info.

File: src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Handlers/Admin/RachisAdminHandler.cs]: fixed

**Comment by karmeli87** [comment]
And this is the exception if the remote was unable to connect the source

```
You are able to reach 'http://127.0.0.1:50633', but he failed to reach you back on http://fake.url:8080.
Please validate the correctness of your 'PublicServerUrl'/'PublicServerUrl.Tcp' configuration and the firewall rules.
Please visit https://ravendb.net/docs/article-page/4.1/csharp/start/installation/setup-wizard#installation--setup-wizard-walkthrough for the RavenDB setup instructions.
```

**Comment by ayende** [comment]
` but he failed` -> `but the *remote node* failed`

**Pull request merged**